Philips.CodeAnalysis.Common/AdditionalFilesHelper.cs

```diff
@@ -84,27 +83,56 @@ public virtual SourceText RetrieveSourceText(string fileName)
 			return null;
 		}
 
-		public virtual ExceptionsOptions LoadExceptionsOptions(SourceText text)
+		public virtual ExceptionsOptions LoadExceptionsOptions(SourceText text, string diagnosticId)
 		{
 			ExceptionsOptions options = new ExceptionsOptions();
 
 			foreach (TextLine textLine in text.Lines)
 			{
 				string line = textLine.ToString();
-				if (line.Contains($@"dotnet_code_quality.{_diagnosticId}.ignore_exceptions_file"))
+				if (line.Contains($@"dotnet_code_quality.{diagnosticId}.ignore_exceptions_file"))
 				{
 					options.IgnoreExceptionsFile = true;
 				}
-				if (line.Contains($@"dotnet_code_quality.{_diagnosticId}.generate_exceptions_file"))
+				if (line.Contains($@"dotnet_code_quality.{diagnosticId}.generate_exceptions_file"))
 				{
 					options.GenerateExceptionsFile = true;
 				}
 			}
 			return options;
 		}
 
-	}
+		/// <summary>
+		/// Get a list of values (comma separated) for the given setting in editorconfig
+		/// </summary>
+		/// <returns></returns>
+		public virtual List<string> GetValuesFromEditorConfig(string diagnosticId, string settingKey)
+		{
+			SourceText lines = RetrieveSourceText(EditorConfig);
+			List<string> values = new List<string>();
 
+			if (lines == null)
+				return values;
+
+			foreach (TextLine textLine in lines.Lines)
+			{
+				string line = textLine.ToString();
+				if (line.Contains($@"dotnet_code_quality.{diagnosticId}.{settingKey}"))
+				{
+					if (line.Contains('='))
+					{
+						string value = line.Substring(line.IndexOf('=') + 1).Trim();
+						foreach (string v in value.Split(','))
```

Comment:
We should handle bogus input to avoid analyzer crashes.  E.g., ends with = ?

Reply:
Testing shows that it also works when the setting ends with =